### PR TITLE
articles view ticket 3

### PR DESCRIPTION
**Pull request by PabloJulianRial**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,12 +1,45 @@
...
 
 
   return (
-    <div>
-      NC News
+    <div className='App'>
+     <Header />
+     <Routes>
+     <Route
+          path="/articles"
```
Comment (NRMSMITH) [inline src/App.jsx]: You don't have a route for '/' which is the default for this webpage - please consider if you want someone to have to click a button to access the ArticleList, or whether you want that to be the first page.

File: src/App.jsx [hunk 2]
```diff
@@ -1,12 +1,45 @@
...
+  useEffect(() => {
+    getArticles().then((response) => {
+      console.log(response)
+      setAllArticles(response);
+    });
+    
+  }, []);
+  useEffect(() => {
+    getTopics().then((response) => {
+      console.log(response)
```
Comment (NRMSMITH) [inline src/App.jsx]: delete your console logs - everything pushed to github should be 'production ready'

File: src/App.jsx [hunk 3]
```diff
@@ -1,12 +1,45 @@
...
+
+
 
 function App() {
+  const [allArticles, setAllArticles] = useState([])
+  const [allTopics, setAllTopics] = useState([])
+
+  useEffect(() => {
+    getArticles().then((response) => {
+      console.log(response)
```
Comment (NRMSMITH) [inline src/App.jsx]: ^ delete

File: src/App.jsx [hunk 4]
```diff
@@ -1,12 +1,45 @@
...
+import ArticlesList from './components/ArticlesList'
+import { Route, Routes } from "react-router-dom";
+import { getArticles } from "./utils";
+import { getTopics } from "./utils";
+
+
 
 function App() {
+  const [allArticles, setAllArticles] = useState([])
+  const [allTopics, setAllTopics] = useState([])
```
Comment (NRMSMITH) [inline src/App.jsx:13]: It might be nice to create some state here that indicates if the data is still loading

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
This looks great so far!

There's a bit of confusion on your Routes - you're fine to have the endpoint /articles appear on a route of '/' - that will definitely improve the UI (I had to search for why nothing was loading and there was no button - I had to type '/articles' at the end of the URL in order to use your website for its intended purpose.

I would also suggest adding a loading stte in order to improve the UI - just in case they have a poor internet connection or something else to that effect.

I would also suggest that the 'Search by topic' is a ticket _much_ later on, and therefore shouldn't appear on this PR! Please ensure from now on that you're not blending tickets together!

File: src/components/ArticlesList.jsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    <div className="articles-list">
+      <label htmlFor="topics-menu">Search by topic:</label>
+      <select
+        onChange={(e) => {
+          setCurrentTopic(e.target.value);
+        }}
+        name="topics-menu"
+        id="topics-menu"
+      >
+        <option value="none"></option>
```
Comment (B7Bprog) [inline src/components/ArticlesList.jsx]: Topics are in Ticket 9, should not be included in this PR.

File: src/components/ArticlesList.jsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        {allTopics.map(( topic ) => {
+          return (
+            <option key={topic} value={topic}>
+              {topic}
+            </option>
+          );
+        })}
+      </select>
+      {allArticles.map((article) => {
+        return (
```
Comment (B7Bprog) [inline src/components/ArticlesList.jsx:12]: All these tags could be extracted into a separate component.

File: src/utils.jsx [hunk 7]
```diff
@@ -0,0 +1,19 @@
+import axios from "axios";
```
Comment (B7Bprog) [inline src/utils.jsx]: Please rename this file to have a .js extension. This is not a React component.

File: src/utils.jsx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+export function getArticles() {
+    return axios.get('https://nc-news-api-iv8w.onrender.com/api/articles')
```
Comment (B7Bprog) [inline src/utils.jsx]: Here try to create an axios instance with a baseURL. For this see NC notes about axios.

File: src/utils.jsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    .then(({data}) =>{
+        return data.articles
+    })
+}
+
+export function getTopics() {
```
Comment (B7Bprog) [inline src/utils.jsx]: This shouldn't be part of this PR.

File: src/components/ArticleOverview.jsx [hunk 10]
```diff
@@ -0,0 +1,14 @@
+const ArticleOverview = ({title, author}) =>{
+
+    console.log({title})
```
Comment (B7Bprog) [inline src/components/ArticleOverview.jsx]: Please remove all logs.

File: src/App.jsx [hunk 11]
```diff
@@ -1,12 +1,54 @@
...
+  const [isLoading, setIsLoading] = useState(true)
+  const [isError, setIsError] = useState(false)
+
+  useEffect(() => {
```
Comment (B7Bprog) [inline src/App.jsx:17]: The 'isLoading' state should be false by default, and set to true here, just before fetching. This way it gets toggled the right way every time you fetch.

File: src/components/ArticlesList.jsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+          <article key={article.title} className="article-item">
```
Comment (B7Bprog) [inline src/components/ArticlesList.jsx]: You could consider utilising the pictures as well for the articles.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Pablo, please see my comments and try to refactor your code, then send us a quick PR with it. 👍

File: src/components/Articlepreview.jsx [hunk 13]
```diff
@@ -0,0 +1,15 @@
+const ArticlePreview = ({article}) =>{
+
+    return(
+        <article key={article.title} className="article-item">
+        <p>Title: {article.title}</p>
+        <p>Topic: {article.topic}</p>
+        <p>Author: {article.author}</p>
+        <p>Posted on: {article.created_at.slice(0, 10)}</p>
+        <img src={article.article_img_url} alt="" />
```
Comment (NRMSMITH) [inline src/components/Articlepreview.jsx]: Please make sure to add an alt text here :)

**Review by NRMSMITH — APPROVED** [review]
Please add alt text to your article preview and then merge 👍
